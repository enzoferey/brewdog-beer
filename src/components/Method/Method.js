import React from "react";
import PropTypes from "prop-types";
import style from "./method.scss";

import MashTemperature from "components/MashTemperature";

const Method = ({
  mash_temp: tempetures,
  fermentation: {
    temp: { value, unit },
  },
  twist,
}) => (
  <div className={style.main}>
    {tempetures.map(({ temp: { value, unit }, duration }, index) => (
      <MashTemperature
        key={`mash-${index}`}
        value={value}
        unit={unit}
        duration={duration}
      />
    ))}
    <p className={style.amount}>{`${value} ${unit}`}</p>
    <p className={style.twist}>{twist}</p>
  </div>
);

const TemperaturePropTypes = PropTypes.shape({
  value: PropTypes.number,
  unit: PropTypes.string,
});

Method.propTypes = {
  mash_temp: PropTypes.arrayOf(
    PropTypes.shape({
      temp: TemperaturePropTypes,
      duration: PropTypes.number,
    })
  ),
  fermentation: PropTypes.shape({
    temp: TemperaturePropTypes,
  }),
  twist: PropTypes.string,
};

export default Method;
