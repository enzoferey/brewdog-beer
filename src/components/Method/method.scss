.temperatures {
  width: 100%;
  position: relative;

  > div + div {
    margin-top: 20px;
  }
}
