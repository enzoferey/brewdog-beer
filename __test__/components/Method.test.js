import React from "react";
import Method from "components/Method";

const method = mocks.beers[0].method;

test("renders correctly", () => {
  testSnapshot(<Method {...method} />);
});
