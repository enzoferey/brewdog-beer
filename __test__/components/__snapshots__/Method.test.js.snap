// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <div
        className={undefined}
    >
        <div
            className={undefined}
        >
            <div
                className={undefined}
            >
                <span>
                    1
                </span>
            </div>
            <img
                alt=""
                className={undefined}
                src="temperature.png"
            />
            <span
                className={undefined}
            >
                64ºC
            </span>
            <span
                className={undefined}
            >
                5s
            </span>
        </div>
    </div>
</div>,
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <img
        alt=""
        className={undefined}
        src="fermentation.png"
    />
    <span
        className={undefined}
    >
        19ºC
    </span>
</div>,
  <div
    className={undefined}
    style={
        Object {
            "flex": 2,
          }
    }
>
    <p
        className={undefined}
    >
        -
    </p>
</div>,
]
`;
